### PR TITLE
Better lasso highlighting

### PR DESCRIPTION
**Pull request by DavidACosgrove**
#### Reference Issue
No issue

#### What does this implement/fix? Explain your changes.
This is a re-do of the lasso highlighting, prompted by the rather late realisation that the results produced on Windows were completely wrong.
The code is now much simpler and doesn't need any of the post-processing of lines and arcs to remove things that shouldn't have been there.

#### Any other comments?
It now gives the same results on Windows and macOS.  Linux hasn't been tested but it gave the same results as macOS previously.
Commit 6f1e277 should say "Most tests now working".  Doh.
Apologies for such a botched first attempt.

### CODE REVIEW

**Comment by greglandrum** [comment]
FWIW: I ran the tests on linux and got no warnings about hash mismatches. I also looked at all the lasso outputs and they seem fine

File: Code/GraphMol/MolDraw2D/DrawMolMCHLasso.cpp [hunk 1]
```diff
@@ -172,35 +171,321 @@ double getLassoWidth(const DrawMolMCH *dm, int atNum, int lassoNum) {
...
+            std::make_pair((arc->ang2_ + arc->ang1_) / 2.0, bangs.size()));
+      } else {
+        double bang = -360.0 + (arc->ang2_ + arc->ang1_ + 360.0) / 2.0;
+        bangs.push_back(std::make_pair(bang, bangs.size()));
+      }
+      arc.reset();
+    }
+  }
+  currArcs.erase(
+      remove_if(currArcs.begin(), currArcs.end(),
```
Comment (greglandrum) [inline Code/GraphMol/MolDraw2D/DrawMolMCHLasso.cpp:411]: This probably should be
```suggestion
      std::remove_if(currArcs.begin(), currArcs.end(),
```

**Review by greglandrum — CHANGES_REQUESTED** [review]
I didn't do a deep review since this is isolated (and complex) code and the tests all look good.

There's one minor suggestion in there just so you don't think I didn't look at the code at all. ;-)

File: Code/GraphMol/MolDraw2D/DrawMolMCHLasso.cpp (hunk 1 above)
Comment (DavidACosgrove) [reply Code/GraphMol/MolDraw2D/DrawMolMCHLasso.cpp:411]: Thanks.  I have pushed that and also a gratuitous grammar fix in one of the comments.
Let's hope it all works better this time.

**Review by greglandrum — APPROVED** [review]
LGTM

**Pull request merged**